Store the name of a mounted bucket in block's metadata (#1208)
awslabs/mountpoint-s3: mountpoint-s3/CHANGELOG.md

### Other changes

* Fix an issue where an interrupt during `readdir` syscall leads to an error. ([#965](https://github.com/awslabs/mountpoint-s3/pull/965))

> * Fix an issue with the missing verification of source bucket of a cache block ([#1208](https://github.com/awslabs/mountpoint-s3/pull/1208))

Reviewer: ```suggestion
* Fix an issue where the source bucket of a shared cache block was not correctly validated ([#1208](https://github.com/awslabs/mountpoint-s3/pull/1208))
```
